stdlib/src/utils/_utf8_validation.mojo

```diff
@@ -0,0 +1,176 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+
```

Comment:
```suggestion
# Reference:
# J. Keiser, D. Lemire, Validating UTF-8 In Less Than One Instruction Per Byte, 
# Software: Practice and Experience 51 (5), 2021
# https://arxiv.org/abs/2010.03090
```

Reply:
I suggest adding a reference to the paper describing the algorithm. If I am a regular programmer and I check this code, I may not understand what is going on right away. Having a paper as a reference could be helpful.

Reply:
Thanks, the references were the main function, but it's true that it makes sense to put them at the top of the file to increase visibility. I just moved them in the latest commit.

Reply:
@gabrieldemarmiesse :-)